logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/log/LogStorageAppender.java

```diff
@@ -178,10 +186,19 @@ public void onActorFailed() {
   }
 
   private void onWriteBufferAvailable() {
-    final BlockPeek blockPeek = new BlockPeek();
-    if (writeBufferSubscription.peekBlock(blockPeek, maxAppendBlockSize, true) > 0) {
-      appendBlock(blockPeek);
-    } else {
+    final var blockPeek = new BlockPeek();
+    final var readBytes = writeBufferSubscription.peekBlock(blockPeek, maxAppendBlockSize, true);
+
+    final var shouldAppendBlock = readBytes > 0;
+    var appendBlockSucceeded = false;
+
+    if (shouldAppendBlock) {
+      appendBlockSucceeded = appendBlock(blockPeek);
+    }
+
+    // Yield the thread, when no bytes are read or
+    // when the block could not be appended.
```

Comment:
:wrench: I think this comment is not necessary. It documents the what, which we can see in the code. It would be useful if we would document the why :)

Reply:
I personally think the comment explains the what and the why 😃 However, I removed the comment.

Reply:
For me it would be more useful why we need to yield. Like without the yield we try in a loop endless after failing to append and no other thread is able to do other work. This will overwhelm the raft thread, since we not react on the backpressure.